### PR TITLE
wifi: esp32s2: add wifi station support

### PR DESCRIPTION
**Pull request by glaubermaroto**
Adds wifi station driver support for the ESP32-S2 SoC.

Brings wifi station sample code support for ESP32-S2.

### CODE REVIEW

**Comment by cfriedt** [comment]
Looks good!

I wonder if it would be hard to integrate either esp32s2 or esp32c3 into the generic wifi sample.

https://docs.zephyrproject.org/latest/samples/net/wifi/README.html

Might make for another good enhancement.

**Review by sylvioalves — APPROVED** [review]

**Comment by sylvioalves** [comment]
> Looks good!
> 
> I wonder if it would be hard to integrate either esp32s2 or esp32c3 into the generic wifi sample.
> 
> https://docs.zephyrproject.org/latest/samples/net/wifi/README.html
> 
> Might make for another good enhancement.

I have this working already. Just need more time to finish some stuff. Will PR soon.

**Comment by glaubermaroto** [comment]
@cfriedt Thanks for the suggestion!

As @sylvioalves  pointed out, he is already addressing that integration in a separate PR.

**Review by uLipe — APPROVED** [review]

File: dts/xtensa/espressif/esp32s2.dtsi [hunk 1]
```diff
@@ -25,6 +25,11 @@
 		};
 	};
 
+	wifi: wifi {
+		compatible = "espressif,esp32-wifi";
+		status = "disabled";
+	};
+
```
Comment (mbolivar-nordic) [inline dts/xtensa/espressif/esp32s2.dtsi:36]: What is this supposed to do? There's no binding for this compatible.

Comment (sylvioalves) [reply dts/xtensa/espressif/esp32s2.dtsi:36]: @mbolivar-nordic I think we can remove all entries regarding this in all Espressif SoCs. Current wifi implementation does not use this binding at all. I'll submit a PR, sounds good removing this whole entry?

Comment (mbolivar-nordic) [reply dts/xtensa/espressif/esp32s2.dtsi:36]: Sure

**Pull request merged**